Tools/AP_Periph/can.cpp

```diff
@@ -1277,6 +1302,41 @@ static void processRx(void)
     }
 }
 
+#ifdef HAL_PERIPH_CAN_MIRROR
+static void processMirror(void)
+{
+    const uint64_t deadline = AP_HAL::native_micros64() + 1000000;
+
+    for (auto &ins : instances) {
+        if (ins.iface == nullptr || ins.mirror_queue == nullptr) { // can't send on a null interface
+            continue;
+        }
+
+        for (uint32_t i = 0; i < ins.mirror_queue->available (); i++) { // limit how long we can loop
```

Comment:
whitespace. Also, please cache ```ins.mirror_queue->available()```so there's zero chance of it growing while you're looping

Reply:
I can cache `ins.mirror_queue->available()` I didn't because it can't grow here. It's only inserted to by the `processRX()` path.